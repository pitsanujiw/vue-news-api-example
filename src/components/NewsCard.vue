<template>
  <v-card class="ma-2" min-width="350" max-width="350">
    <v-img
      height="250"
      class="white--text align-end"
      transition="src"
      :src="newsArticle.urlToImage"
    >
    </v-img>
    <v-card-actions class="d-flex justify-end">
      <news-detail :newsArticle="newsArticle" />
    </v-card-actions>
    <v-card-title>{{ newsArticle.title }}</v-card-title>
    <v-card-subtitle
      ><v-chip class="mt-2">{{
        newsArticle.source.name
      }}</v-chip></v-card-subtitle
    >
    <v-card-text>
      <p>{{ newsArticle.author || "-" }}</p>
      <v-divider></v-divider>

      <div>
        {{ newsArticle.description || '-' }}
      </div>
    </v-card-text>
  </v-card>
</template>

<script>
import newsDetail from './NewsDetail.vue';

export default {
  name: 'newsCard',
  props: {
    newsArticle: {
      type: Object,
    },
  },
  data: () => ({
    news: undefined,
  }),
  methods: {
  },
  components: {
    newsDetail,
  },
};
</script>

<style>
</style>
