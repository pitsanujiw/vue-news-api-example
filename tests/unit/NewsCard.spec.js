import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import NewsCard from '../../src/components/NewsCard.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('NewsCard', () => {
  const modules = {
    getters: {},
  };
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
    store = new Vuex.Store({
      modules: {
        newModules: modules,
      },
    });
  });

  it('Renders NewsCard Component', () => {
    const wrapper = shallowMount(NewsCard, { store, localVue, vuetify });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
