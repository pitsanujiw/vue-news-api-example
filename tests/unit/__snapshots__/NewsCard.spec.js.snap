// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard Renders NewsCard Component 1`] = `
<v-card-stub loaderheight="4" tag="div" maxwidth="350" minwidth="350" class="ma-2">
  <v-img-stub height="250" options="[object Object]" position="center center" src="__URL__" transition="src" class="white--text align-end"></v-img-stub>
  <v-card-actions-stub tag="div" class="d-flex justify-end">
    <news-detail-stub newsarticle="[object Object]"></news-detail-stub>
  </v-card-actions-stub>
  <v-card-title-stub tag="div">__TITLE__</v-card-title-stub>
  <v-card-subtitle-stub tag="div">
    <v-chip-stub activeclass="" tag="span" active="true" closeicon="$delete" closelabel="$vuetify.close" filtericon="$complete" class="mt-2">__SOURCE_NAME__</v-chip-stub>
  </v-card-subtitle-stub>
  <v-card-text-stub tag="div">
    <p>__AUTHOR__</p>
    <v-divider-stub></v-divider-stub>
    <div>
      __DESCRIPTION__
    </div>
  </v-card-text-stub>
</v-card-stub>
`;
